projects/plugins/protect/src/class-scan-history.php

```diff
@@ -0,0 +1,383 @@
+<?php
+/**
+ * Class to handle the Scan Status of Jetpack Protect
+ *
+ * @package automattic/jetpack-protect-plugin
+ */
+
+namespace Automattic\Jetpack\Protect;
+
+use Automattic\Jetpack\Connection\Client;
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
+use Jetpack_Options;
+use WP_Error;
+
+/**
+ * Class that handles fetching of threats from the Scan API
+ */
+class Scan_History {
+	/**
+	 * Scan endpoint
+	 *
+	 * @var string
+	 */
+	const SCAN_HISTORY_API_BASE = '/sites/%d/scan/history';
+
+	/**
+	 * Name of the option where history is stored
+	 *
+	 * @var string
+	 */
+	const OPTION_NAME = 'jetpack_scan_history';
+
+	/**
+	 * Name of the option where the timestamp of the history is stored
+	 *
+	 * @var string
+	 */
+	const OPTION_TIMESTAMP_NAME = 'jetpack_scan_history_timestamp';
+
+	/**
+	 * Time in seconds that the cache should last
+	 *
+	 * @var int
+	 */
+	const OPTION_EXPIRES_AFTER = 300; // 5 minutes.
+
+	/**
+	 * Memoization for the current history
+	 *
+	 * @var null|History_Model
+	 */
+	public static $history = null;
+
+	/**
+	 * Checks if the current cached history is expired and should be renewed
+	 *
+	 * @return boolean
+	 */
+	public static function is_cache_expired() {
+		$option_timestamp = get_option( static::OPTION_TIMESTAMP_NAME );
+
+		if ( ! $option_timestamp ) {
+			return true;
+		}
+
+		return time() > (int) $option_timestamp;
+	}
+
+	/**
+	 * Checks if we should consider the stored cache or bypass it
+	 *
+	 * @return boolean
+	 */
+	public static function should_use_cache() {
+		return ! ( ( defined( 'JETPACK_PROTECT_DEV__BYPASS_CACHE' ) && JETPACK_PROTECT_DEV__BYPASS_CACHE ) );
+	}
+
+	/**
+	 * Gets the current cached history
+	 *
+	 * @return bool|array False if value is not found. Array with values if cache is found.
+	 */
+	public static function get_from_options() {
+		return maybe_unserialize( get_option( static::OPTION_NAME ) );
+	}
+
+	/**
+	 * Updated the cached history and its timestamp
+	 *
+	 * @param array $history The new history to be cached.
+	 * @return void
+	 */
+	public static function update_history_option( $history ) {
+		// TODO: Sanitize $history.
+		update_option( static::OPTION_NAME, maybe_serialize( $history ) );
+		update_option( static::OPTION_TIMESTAMP_NAME, time() + static::OPTION_EXPIRES_AFTER );
+	}
```

Comment:
There's a relevant `TODO` here already, it would be nice if we had a stronger type for `$history`. Perhaps using the model class?

Reply:
Ah, I see. Since we're caching the API response, serializing it is likely all we need to do here. We're sanitizing the data via `normalize_api_data`, so this LGTM :+1: